### PR TITLE
added smoother error code bt node 

### PR DESCRIPTION
**Pull request by jwallace42**
Updated documentation for the new smoother error code bt node

### CODE REVIEW

File: tutorials/docs/adding_smoother.rst [hunk 1]
```diff
@@ -72,17 +72,27 @@ A given behavior tree will have a line:
 
 .. code-block:: xml
```
Comment (SteveMacenski) [inline tutorials/docs/adding_smoother.rst:73]: I think this is missing the configuration guide page (along with the others)

File: tutorials/docs/adding_smoother.rst [hunk 2]
```diff
@@ -72,17 +72,27 @@ A given behavior tree will have a line:
...
+      <ComputePathToPose goal="{goal}" path="{path}" planner_id="GridBased" error_code_id="{compute_path_error_code}"/>
+      <SmoothPath unsmoothed_path="{path}" smoothed_path="{path}" error_code_id={smoother_error_code}"/>
     </Sequence>
 
+If you wish to have recoveries for the smoother error codes:
+
+.. code-block:: xml 
+
+    <Sequence name= "TryToResolveSmootherErrorCodes">
+      <WouldASmootherRecoveryHelp error_code="{would_a_smoother_recovery_help}">
```
Comment (SteveMacenski) [inline tutorials/docs/adding_smoother.rst:91]: `smoother_error_code`?

File: tutorials/docs/adding_smoother.rst [hunk 3]
```diff
@@ -72,17 +72,27 @@ A given behavior tree will have a line:
...
 
 This line calls the planner server and return a path to the ``path`` blackboard variable in the behavior tree. We are  going to replace that line with the following to compute the path, smooth the path, and finally replace the ``path`` blackboard variable with the new smoothed path that the system will now interact with:
 
 .. code-block:: xml
 
     <Sequence name="ComputeAndSmoothPath">
-      <ComputePathToPose goal="{goal}" path="{path}" planner_id="GridBased"/>
-      <SmoothPath unsmoothed_path="{path}" smoothed_path="{path}"/>
+      <ComputePathToPose goal="{goal}" path="{path}" planner_id="GridBased" error_code_id="{compute_path_error_code}"/>
+      <SmoothPath unsmoothed_path="{path}" smoothed_path="{path}" error_code_id={smoother_error_code}"/>
```
Comment (SteveMacenski) [inline tutorials/docs/adding_smoother.rst:83]: Missing quotes

File: tutorials/docs/adding_smoother.rst [hunk 4]
```diff
@@ -72,17 +72,27 @@ A given behavior tree will have a line:
...
     </Sequence>
 
+If you wish to have recoveries for the smoother error codes:
```
Comment (SteveMacenski) [inline tutorials/docs/adding_smoother.rst:86]:
```suggestion
If you wish to have recoveries for the smoother error codes, such as triggering the system recoveries branch of a behavior tree:
```

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

File: tutorials/docs/adding_smoother.rst (hunk 1 above)
Comment (jwallace42) [reply tutorials/docs/adding_smoother.rst:73]: Added this in!

File: configuration/packages/bt-plugins/conditions/AreErrorCodesPresent.rst [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+  Description
+    	The set of error codes you wish to compare against the active error code. 
+
+Example
+-------
+
+.. code-block:: xml
+
+    <AreErrorCodesPresent error_code="{error_code}" error_codes_to_check="{error_codes_to_check}"/>
```
Comment (SteveMacenski) [inline configuration/packages/bt-plugins/conditions/AreErrorCodesPresent.rst:40]: Can you show some specific codes? Maybe 2 entries: 1 with a BT variable containing them (like you have) and another with a list of numbers that works (e.g. `403, 205, 302,` etc)?

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

File: configuration/packages/bt-plugins/conditions/AreErrorCodesPresent.rst (hunk 5 above)
Comment (SteveMacenski) [reply configuration/packages/bt-plugins/conditions/AreErrorCodesPresent.rst:40]: I feel like that'll be a common application of this node

**Review by SteveMacenski — APPROVED** [review]

**Pull request merged**